### ISSUE TITLE
Uniformize the columns returned in a MVT for Mapnik and St_AsMVT

### ISSUE THREAD

**Issue by Algunenano** [CartoDB/Windshaft#628]
If I instantiate a map through Builder and request an MVT file by just replacing the extension, `.png` with `.mvt`, I get 2 different files depending whether `mvt::usePostGIS` is enabled or not. With it disabled, Mapnik is used and all the columns in the map query are returned. With it enabled only `the_geom_webmercator` is returned.

Since I understand we want to be able to switch between the 2 of them flawlessly this is certainly a major issue, as anything that is currently using the MVT in production and relying on other columns to be returned will be broken once the direct ST_AsMVT is enabled.

I'm proposing changing the default for ST_AsMVT to return all columns in the tile query to match the current behaviour.  @davidmanzanares @rochoa , since I see you were the ones that implemented it, what do you think?

**Comment by Algunenano** [comment]
WIP in https://github.com/Algunenano/Windshaft/commit/0fa06ca84c55bd1e9c59bcd55329addf21f1ef64

**Comment by rochoa** [comment]
I want to think about it.

My current memory dump:
1. Sending all the columns is inefficient.
1. Breaking the existing behaviour is unacceptable
1. We could send all columns depending on the MapConfig's options, e.g. if cartocss is present, we send all columns for MVT tiles.
1. We could send all columns always, except if the MapConfig has a `columns` option requesting only specific columns.
1. We could create a `/api/v2/map` endpoint where we change the existing behaviour, not breaking existing clients.

**Comment by davidmanzanares** [comment]
I agree with @rochoa, but I would add:

6. Users of CARTO VL won't need this feature on Windshaft to get the same result (we tweak the SQL query to just get the required columns).
7. Developers that know the cost of downloading all columns can get the same result by changing the SQL query.
8. Supporting this extra feature is costly.

So, in my opinion, storing all columns is the better option since I don't think its advantages worth the cost.

**Comment by Algunenano** [comment]
> Sending all the columns is inefficient.

It's what we are currently doing with mapnik. If the pg-mvt renderer only returns the geometry by default, any map currently using mvt's and expecting the columns to be included will break once we switch `usePostgis` to on.

> Breaking the existing behaviour is unacceptable

I haven't any usage of `layer.options.columns` anywhere except Windshaft own tests, not even in Windshaft-cartodb. Also it wasn't documented.
Windshaft-cartodb has some examples with `layer.options.attributes.columns` but that's intended for attributes backend, not mvts.

> We could send all columns always, except if the MapConfig has a columns option requesting only specific columns.

I agree that's a nice feature to have, but we'd need to define the API and make it consistent between renderers. Should it be `layer.options.columns`, `layer.options.attributes.columns` or other? 

> We could create a /api/v2/map endpoint where we change the existing behaviour, not breaking existing clients.

As I mentioned before, it wasn't documented and I couldn't find anyone using it. I've tried Kibana but I couldn't find a request with `render-mvt` ending in a 200. Maybe I'm not looking correctly :man_shrugging: 

Also, we have `[:use_postgis] = false` in production, so I'm pretty sure we won't break any customer map, which you can't be certain if you have different behaviour between the renderers.

> Users of CARTO VL won't need this feature on Windshaft to get the same result (we tweak the SQL query to just get the required columns).
Developers that know the cost of downloading all columns can get the same result by changing the SQL query.

That's true but, to be honest, if mvt werent' public yet I'd rather use Ochoa's proposal of disabling everything and requesting the user to opt-in per column. Nevertheless I'd rather have a consistent result between mapnik and pg-mvt renderer, and mapnik is already returning all columns with known types.

TLDR: I'd rather have a smooth transition between mapnik and pg-mvt renderers than an opt-in attribute system.

**Comment by rochoa** [comment]
> TLDR: I'd rather have a smooth transition between mapnik and pg-mvt renderers than an opt-in attribute system.

^^ I agree on this, and, honestly, it's the only viable option.

I was just memory-dumping my ideas.

**Comment by Algunenano** [comment]
Done in #632
